### PR TITLE
implement rspec, cucumber and shoulda ( shoulda-matchers)

### PR DESCRIPTION
**Pull request by vuducdat**

### CODE REVIEW

File: config/cucumber.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<%
+rerun = File.file?('rerun.txt') ? IO.read('rerun.txt') : ""
+rerun_opts = rerun.to_s.strip.empty? ? "--format #{ENV['CUCUMBER_FORMAT'] || 'progress'} features" : "--format #{ENV['CUCUMBER_FORMAT'] || 'pretty'} #{rerun}"
+std_opts = "--format #{ENV['CUCUMBER_FORMAT'] || 'pretty'} --strict --tags ~@wip"
+%>
+default: <%= std_opts %> features
+wip: --tags @wip:3 --wip features
+rerun: <%= rerun_opts %> --format rerun --out rerun.txt --strict --tags ~@wip
```
Comment (phuongbk) [inline config/cucumber.yml]: Change all ' to "

File: lib/tasks/cucumber.rake [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+  task :stats => 'cucumber:statsetup'
+rescue LoadError
+  desc 'cucumber rake task not available (cucumber not installed)'
+  task :cucumber do
+    abort 'Cucumber rake task is not available. Be sure to install cucumber as a gem or plugin'
+  end
+end
+
+end
```
Comment (phuongbk) [inline lib/tasks/cucumber.rake]: Change all ' to "

File: script/cucumber [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env ruby
+
+vendored_cucumber_bin = Dir["#{File.dirname(__FILE__)}/../vendor/{gems,plugins}/cucumber*/bin/cucumber"].first
+if vendored_cucumber_bin
+  load File.expand_path(vendored_cucumber_bin)
+else
+  require 'rubygems' unless ENV['NO_RUBYGEMS']
+  require 'cucumber'
+  load Cucumber::BINARY
+end
```
Comment (phuongbk) [inline script/cucumber]: change all ' to "

File: Gemfile [hunk 4]
```diff
@@ -59,6 +59,15 @@ group :development, :test do
   # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
   gem "spring"
   gem "rspec-rails", "~> 3.0"
+  gem "rspec-collection_matchers"
+  gem "rspec-mocks"
```
Comment (yuyuvn) [inline Gemfile]: rspec-mocks is in rspec-rails's dependcy so you don't need it in Gemfile

**Pull request merged**